### ISSUE TITLE
Support asset_host as a Proc

### ISSUE THREAD

**Issue by lewisf** [ElMassimo/vite_ruby#202]
Rails allows passing in a `proc` for asset host -- something like this:

```
ActionController::Base.asset_host = Proc.new { |source|
  "http://assets#{Digest::MD5.hexdigest(source).to_i(16) % 2 + 1}.example.com"
}
```

Passing in a Proc here currently breaks the helper template tags since the Proc case is not covered

**Comment by ElMassimo** [comment]
Since dynamic values for `asset_host` rely on the context of the current request, there's no way to pass that to Vite.

Usually, [the asset host will be used as the base](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/src/config.ts#L108) for Vite.

A reasonable behavior when `asset_host` is a `Proc` would be not to expose `asset_host` to `vite-plugin-ruby`, so that it uses `/` as the base.

---

In order to better inform my decisions, what is your use case for a dynamic `asset_host`?

**Comment by lewisf** [comment]
Hey @ElMassimo -- I'll give this a look

We use ActiveAdmin, which serves it's own JS/CSS from the Rails web server. In order to ensure that that works, we've done something like:

```
    config.asset_host = proc { |source|
      if source.start_with?('/assets/active_admin')
        'localhost:3000'
      else
        'localhost:3036'
      end
    }
```

**Comment by ElMassimo** [comment]
Gotcha, since for development `asset_host` is typically not relevant, I think the fix I proposed should work for your use case as well.

**Comment by ElMassimo** [comment]
Fixed in `vite_ruby-3.1.1` and `vite_rails-3.0.8`, you can upgrade both by running `bin/vite upgrade`.
